src/gui/tray/UserLine.qml

```diff
@@ -167,10 +180,19 @@ MenuItem {
                                 color: parent.parent.hovered ? Style.lightHover : "transparent"
                             }
                         }
+
+                        Accessible.role: Accessible.Button
+                        Accessible.name: isConnected ? qsTr("Log out") : qsTr("Log in")
+
+                        onPressed: {
+                            isConnected ? UserModel.logout(index) : UserModel.login(index)

```

Comment:
I find constructs like this reduce readability in imperative code like this. Could we have a simple if instead?
```
if (isConnected) {
    UserModel.logout(index)
} else {
   UserModel.login(index)
}

Reply:
QML is declarative JS in most manners - structures like `if` branches are only possible in functions or as property inlines, thus I chose the ternary here to keep it within the object definition.

If you still want that, I'd implement it as a call to a separate function performLoginLogoutAction. I'd consider this bloat - what about you?

Reply:
> QML is declarative JS in most manners

This is a misconception. This is true *only* for property bindings, not for signal handlers. Those are functions not saying their name. And actually Qt 5.15 (rightfully) mark the "onFoo:" syntax deprecated for those to reduce the confusion (and I can smell it'll be outright forbidden in Qt 6 in favor of the new explicit function syntax).

> If you still want that, I'd implement it as a call to a separate function performLoginLogoutAction. I'd consider this bloat - what about you?

I consider it bloat indeed. I'd rather have the if in the body of that function. :-)

Reply:
Ahhh, makes totally sense, never realised that indeed. Change coming in.